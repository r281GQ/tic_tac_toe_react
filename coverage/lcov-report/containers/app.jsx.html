<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/app.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">containers</a> app.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.31% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.31% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/13</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
&nbsp;
import * as actions from './../store/action_creators/board';
import ProgressBar from './../components/progress_bar';
import GameEndedDialog from './../components/game_ended_dialog';
import Board from './../components/board';
import Header from './../components/header';
&nbsp;
class App extends Component {
  constructor(props) {
    super(props);
    this._createNewGame = this._createNewGame.bind(this);
    this._makeHumanMove = this._makeHumanMove.bind(this);
  }
&nbsp;
  _makeHumanMove(number) {
    return () =&gt; {
      this.props.makeHumanMove(number);
      this.props.checkForGameState();
      this.props.makeComputerMove();
      this.props.checkForGameState();
    };
  }
&nbsp;
  _createNewGame() {
    this.props.newGame();
  }
&nbsp;
  render() {
    return (
      &lt;div&gt;
        &lt;Header createNewGameHandler={this._createNewGame} /&gt;
        &lt;GameEndedDialog
          isModalOpen={this.props.isModalOpen}
          createNewGameHandler={this._createNewGame}
          message={this.props.message}
        /&gt;
        &lt;ProgressBar isCalculating={this.props.isCalculating} /&gt;
        &lt;Board
          board={this.props.board}
          makeHumanMove={this._makeHumanMove}
          playerSign={this.props.playerSign}
          computerSign={this.props.computerSign}
        /&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
App.propTypes = {
  board: PropTypes.shape({
    1: PropTypes.string,
    2: PropTypes.string,
    3: PropTypes.string,
    4: PropTypes.string,
    5: PropTypes.string,
    6: PropTypes.string,
    7: PropTypes.string,
    8: PropTypes.string,
    9: PropTypes.string
  }).isRequired,
  makeHumanMove: PropTypes.func.isRequired,
  makeComputerMove: PropTypes.func.isRequired,
  checkForGameState: PropTypes.func.isRequired,
  playerSign: PropTypes.string.isRequired,
  computerSign: PropTypes.string.isRequired,
  isCalculating: PropTypes.bool,
  newGame: PropTypes.func.isRequired,
  isModalOpen: PropTypes.bool,
  message: PropTypes.string
};
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    board: state.board.board,
    isCalculating: state.board.settings.isCalculating,
    playerSign: state.board.settings.playerSign,
    computerSign: state.board.settings.computerSign,
    isModalOpen: state.board.settings.isModalOpen,
    message: state.board.settings.message
  };
};
&nbsp;
export default connect(mapStateToProps, actions)(App);
export { App as PureApp };
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Aug 20 2017 12:24:54 GMT+0100 (BST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
