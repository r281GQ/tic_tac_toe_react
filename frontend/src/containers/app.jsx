import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from './../store/action_creators/board';
import ProgressBar from './../components/progress_bar';
import GameEndedDialog from './../components/game_ended_dialog';
import Board from './../components/board';
import Header from './../components/header';

class App extends Component {
  constructor(props) {
    super(props);
    this._createNewGame = this._createNewGame.bind(this);
    this._makeHumanMove = this._makeHumanMove.bind(this);
  }

  _makeHumanMove(number) {
    return () => this.props.makeHumanMove(number);
  }

  _createNewGame() {
    this.props.newGame();
  }

  render() {
    return (
      <div>
        <Header createNewGameHandler={this._createNewGame} />
        <GameEndedDialog
          isModalOpen={this.props.isModalOpen}
          createNewGameHandler={this._createNewGame}
          message={this.props.message}
        />
        <ProgressBar isCalculating={this.props.isCalculating} />
        <Board
          board={this.props.board}
          makeHumanMove={this._makeHumanMove}
          playerSign={this.props.playerSign}
          computerSign={this.props.computerSign}
        />
      </div>
    );
  }
}

App.propTypes = {
  board: PropTypes.shape({
    1: PropTypes.string,
    2: PropTypes.string,
    3: PropTypes.string,
    4: PropTypes.string,
    5: PropTypes.string,
    6: PropTypes.string,
    7: PropTypes.string,
    8: PropTypes.string,
    9: PropTypes.string
  }).isRequired,
  makeHumanMove: PropTypes.func.isRequired,
  playerSign: PropTypes.string.isRequired,
  computerSign: PropTypes.string.isRequired,
  isCalculating: PropTypes.bool,
  newGame: PropTypes.func.isRequired,
  isModalOpen: PropTypes.bool,
  message: PropTypes.string
};

const mapStateToProps = state => {
  return {
    board: state.board.board,
    isCalculating: state.board.settings.isCalculating,
    playerSign: state.board.settings.playerSign,
    computerSign: state.board.settings.computerSign,
    isModalOpen: state.board.settings.isModalOpen,
    message: state.board.settings.message
  };
};

export default connect(mapStateToProps, actions)(App);
