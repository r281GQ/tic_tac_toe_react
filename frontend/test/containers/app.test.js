import React from 'react';
import { shallow } from 'enzyme';
import { PureApp } from './../../src/containers/app';
import sinon from 'sinon';

describe('App test', () => {
  it('should render', () => {
    const spyNewGame = sinon.spy();
    const spyMakeHumanMove = sinon.spy();
    const spyComputerMove = sinon.spy();
    const spyCheckForGameState = sinon.spy();

    const component = shallow(
      <PureApp
        board={{
          1: '',
          2: '',
          3: '',
          4: '',
          5: '',
          6: '',
          7: '',
          8: '',
          9: ''
        }}
        makeHumanMove={spyMakeHumanMove}
        makeComputerMove={spyComputerMove}
        checkForGameState={spyCheckForGameState}
        playerSign="x"
        computerSign="o"
        isCalculating={false}
        isModalOpen={false}
        message=""
        newGame={spyNewGame}
      />
    );

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();

    component.instance()._createNewGame();
    component.instance()._makeHumanMove(1)();

    expect(spyMakeHumanMove.calledWith(1)).toBeTruthy();
    expect(spyNewGame.calledOnce).toBeTruthy();
    expect(spyComputerMove.calledOnce).toBeTruthy();
    expect(spyCheckForGameState.calledTwice).toBeTruthy();
  });
});
