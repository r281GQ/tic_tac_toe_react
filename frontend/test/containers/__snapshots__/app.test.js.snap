// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should render 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      Tic Tac Toe
    </title>
    <link
      href="mock-file"
      rel="icon"
    />
  </HelmetWrapper>
  <Header
    createNewGameHandler={[Function]}
  />
  <GameEndedDialog
    createNewGameHandler={[Function]}
    isModalOpen={false}
    message=""
  />
  <ProgressBar
    isCalculating={false}
  />
  <Board
    board={
      Object {
        "1": "",
        "2": "",
        "3": "",
        "4": "",
        "5": "",
        "6": "",
        "7": "",
        "8": "",
        "9": "",
      }
    }
    computerSign="o"
    makeHumanMove={[Function]}
    playerSign="x"
  />
</div>
`;
